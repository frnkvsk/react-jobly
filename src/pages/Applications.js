import React, {useState, useEffect, useContext} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import JobCard from '../components/JobCard';
import Search from '../components/SearchBar';
import { Container } from '@material-ui/core';
import { getJobs } from '../api/JoblyApi';
import { SearchContext } from '../context/SearchContext';
import { AuthContext } from '../context/AuthContext';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'flex-start',
    minHeight: '100vh',
    fontSize: '28px',
    maxWidth: 'lg',
    
  },
  form: {
    display: 'flex',
    width: '100%',
    margin: '30px 0 20px 0',
  },
  input: {
    width: '100%'
  },
  card: {
    display: 'flex',
    justifyContent: 'space-between',
    width: '100%',
    padding: '20px 0 20px 20px',
    margin: '10px 0 10px 0',
    border: '1px solid lightgray',
  },
  info: {
    width: '80%',
  },
  icon: {
    fontSize: '40px',
    margin: '30px 20px 0 0',
  },
}));

export default function Applications() {
  const classes = useStyles();
  const searchContext = useContext(SearchContext);
  const [jobs, setJobs] = useState([]);
  const auth = useContext(AuthContext);
  const params = {search: searchContext.searchState.search, _token: auth.authState.token}; 
  
  useEffect(() => {
    const getValue = async () => {
      const value = await getJobs(params);
<<<<<<< HEAD
      setJobs(value.jobs.filter(job => auth.authState.userInfo.jobs.includes(job.id) ) );
=======
      if(value) {
        setJobs(value.jobs.filter(job => auth.authState.userInfo.jobs.includes(job.id) ) );
      }      
>>>>>>> e94462676a209fbe08e3ccc29a60c1d58665523b
    }
    getValue();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [params.search, auth.authState.userInfo.jobs]);

  return (
    <Container className={classes.root}>
      <Search nextPage={"jobs"}/> 
      {jobs.length ? jobs.map(job => <JobCard key={job.id} job={job} /> ) : <h3>You have no applications.</h3>}
    </Container>
  );
}
