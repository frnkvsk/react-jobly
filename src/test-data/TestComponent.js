// import React, { useContext } from 'react';
// import { AuthContext } from '../context/AuthContext';

// const TestComponent = () => {
//   const auth = useContext(AuthContext);
//   const setAuth2 = async () =>
//     await auth.setAuthState({
//       token: 'testtoken',
//       userInfo: {
//       username: 'testuser',
//       first_name: 'testfirst',
//       last_name: 'testlast',
//       email: 'test@test.com',
//       photo_url: null
//     }});
//   (async () => await setAuth2())();
//   console.log('-------auth.authState',auth.authState)
//   return (
//     <div>
//       <div>{auth.authState.userInfo.username}</div>
//       <div>{auth.authState.userInfo.first_name}</div>
//       <div>{auth.authState.userInfo.last_name}</div>
//       <div>{auth.authState.userInfo.email}</div>
//       <div>{auth.authState.userInfo.photo_url}</div>
//     </div>
//   )
// }
// export default TestComponent;

