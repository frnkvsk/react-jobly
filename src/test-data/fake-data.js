const testData = {
  jobs: [
    {
    company_handle: "test-company-a",
    equity: 0.115,
    id: 1,
    salary: 11111,
    state: null,
    title: "Editor, magazine features",
    },
      {
      company_handle: "test-company-b",
      equity: 0.22,
      id: 2,
      salary: 22222,
      state: null,
      title: "Coder",
    },
      {
      company_handle: "test-company-c",
      equity: 0.33,
      id: 3,
      salary: 33333,
      state: null,
      title: "Front-end Dev",
    },
  ],
  companies: [
    {
      description: "Test description 1",
      handle: "test-company-1",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 1",
    },
    {
      description: "Test description 2",
      handle: "test-company-2",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 2",
    },
    {
      description: "Test description 3",
      handle: "test-company-3",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 3",
    },
  ],
}

export { testData }