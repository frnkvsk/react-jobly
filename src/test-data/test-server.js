import { rest } from 'msw';
import { setupServer } from 'msw/node';

let server;

const setServer = async (url, status, message='') => {
  if(server) {
    server.close();
    server.resetHandlers();
  }
  server = await setupServer(
    rest.get(url, (req, res, ctx) => {
      return res(
        ctx.status(status),
        ctx.json(message)
      )
    })
  );
}
  

beforeAll(() => server ? server.listen() : null);
afterAll(() => server ? server.close() : null);
afterEach(() => server ? server.resetHandlers() : null);

export { setServer, server, rest }
