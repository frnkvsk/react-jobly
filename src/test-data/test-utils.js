// test-utils.js
import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { SearchProvider } from '../context/SearchContext';
import { AuthProvider } from '../context/AuthContext';
import { PageCountProvider } from '../context/PageCountContext';

const AllTheProviders = (page) => {
  return render(
    <BrowserRouter>
      <AuthProvider>
        <SearchProvider>
          <PageCountProvider>
            {page}
          </PageCountProvider>
        </SearchProvider>
      </AuthProvider> 
    </BrowserRouter>
    
  );
}

// re-export everything
export * from '@testing-library/react';

// override render method
export { AllTheProviders as render }
