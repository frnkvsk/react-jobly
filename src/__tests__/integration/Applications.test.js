import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '../../test-data/test-utils';
import Applications from '../../pages/Applications';

// smoke and appearance tests
test('renders Applications ', async() => {
  render(<Applications />);   
  
  expect(screen.getByPlaceholderText(/Search for jobs/i)).toBeInTheDocument();

  expect(screen.getByRole('button', {name: /search/i})).toBeInTheDocument();

  expect(screen.getByText(/You have no applications./i)).toBeInTheDocument();
});


// snapshot test
// test('matches snapshot', () => {
//   const { asFragment } = render(<Applications />);
//   expect(asFragment()).toMatchSnapshot();
// });