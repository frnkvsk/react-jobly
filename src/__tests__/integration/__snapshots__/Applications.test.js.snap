// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-root-13 MuiContainer-maxWidthLg"
  >
    <form
      autocomplete="off"
      class="makeStyles-form-19"
      novalidate=""
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-20"
        variant="outlined"
      >
        <input
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="searchInput"
          placeholder="Search for jobs"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-21 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-22"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Search
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
    <h3>
      You have no applications.
    </h3>
  </div>
</DocumentFragment>
`;
